scripts/sagas/bucket.js

```diff
@@ -0,0 +1,111 @@
+import { updatePath } from "redux-simple-router";
+import { call, take, fork, put } from "redux-saga/effects";
+
+import {
+  COLLECTION_LOAD_REQUEST,
+  COLLECTION_CREATE_REQUEST,
+  COLLECTION_UPDATE_REQUEST,
+  COLLECTION_DELETE_REQUEST,
+} from "../constants";
+import { getClient } from "../client";
+import { notifySuccess, notifyError } from "../actions/notifications";
+import { collectionBusy, collectionLoadSuccess } from "../actions/collection";
+import { listBuckets } from "./session";
+
+
+function getBucket(bid) {
+  return getClient().bucket(bid);
+}
+
+function getCollection(bid, cid) {
+  return getBucket(bid).collection(cid);
+}
+
+export function* loadCollection(bid, cid) {
+  try {
+    const coll = getCollection(bid, cid);
+    yield put(collectionBusy(true));
+    const {data} = yield call([coll, coll.getAttributes]);
+    yield put(collectionLoadSuccess({
+      ...data,
+      bucket: bid,
+      label: `${bid}/${cid}`,
+    }));
+  } catch(error) {
+    yield put(notifyError(error));
+  } finally {
+    yield put(collectionBusy(false));
+  }
+}
+
+export function* createCollection(bid, collectionData) {
+  try {
+    const {name, schema, uiSchema, displayFields} = collectionData;
+    const bucket = getBucket(bid);
+    yield call([bucket, bucket.createCollection], name, {
+      data: {uiSchema, schema, displayFields},
+    });
+    yield put(updatePath(`/buckets/${bid}/collections/${name}`));
+    yield put(notifySuccess("Collection created."));
+    yield call(listBuckets);
+  } catch(error) {
+    yield put(notifyError(error));
+  }
+}
+
+export function* updateCollection(bid, cid, collectionData) {
+  try {
+    const coll = getCollection(bid, cid);
+    const {data} = yield call([coll, coll.setMetadata], collectionData);
+    yield put(collectionLoadSuccess({
+      ...data,
+      bucket: bid,
+      label: `${bid}/${cid}`,
+    }));
+    yield put(notifySuccess("Collection properties updated."));
+  } catch(error) {
+    yield put(notifyError(error));
+  }
+}
+
+export function* deleteCollection(bid, cid) {
+  try {
+    const bucket = getBucket(bid);
+    yield call([bucket, bucket.deleteCollection], cid);
+    yield put(updatePath(""));
+    yield put(notifySuccess("Collection deleted."));
+    yield call(listBuckets);
+  } catch(error) {
+    yield put(notifyError(error));
+  }
+}
+
+// Watchers
+
+export function* watchCollectionLoad() {
+  while(true) { // eslint-disable-line
+    const {bid, cid} = yield take(COLLECTION_LOAD_REQUEST);
+    yield fork(loadCollection, bid, cid);
+  }
+}
+
+export function* watchCollectionCreate() {
+  while(true) { // eslint-disable-line
+    const {bid, collectionData} = yield take(COLLECTION_CREATE_REQUEST);
+    yield fork(createCollection, bid, collectionData);
```

Comment:
It basically schedule the execution of the saga in a non-blocking fashion. From the [docs](https://yelouafi.github.io/redux-saga/docs/api/index.html#forkfn-args):

> The middleware doesn't suspend the Generator while waiting for the result of fn. Instead as soon as fn is invoked, the Generator resumes immediately.